mlrun/serving/routers.py

```diff
@@ -710,6 +823,37 @@ def validate(self, request):
                 raise Exception('Expected "inputs" to be a list')
         return request
 
+    def _normalize_weights(self, weights_dict: Dict[str, float]):
+        """
+        Normalized all the weights such that abs(weights_sum - 1.0) <= 0.001
+        and adding 0 weight to all the routes that doesn't appear in the dict.
+        If weights_dict is None the function returns equal weights to all the routes.
+
+        :param weights_dict: weights dictionary {<model_name>: <wight>}
+
+        :return: Normalized weights dictionary
+        """
+        if weights_dict is None:
+            num_of_models = len([*self.routes.keys()])
+            return dict(zip([*self.routes.keys()], [1 / num_of_models] * num_of_models))
+        weights_values = [*weights_dict.values()]
+        weights_sum = np.sum(weights_values)
+        if abs(weights_sum - 1.0) <= 0.001:
```

Comment:
weights sum must be smaller than one, otherwise there was some problem...
```suggestion
        if 1.0 - weights_sum <= 1e-5:
```